src/main/java/uk/gov/pay/directdebit/common/clients/GoCardlessClientFacade.java

```diff
@@ -50,12 +50,11 @@ public GoCardlessCustomer createCustomerBankAccount(MandateExternalId mandateExt
         return customer;
     }
 
-    public GoCardlessMandate createMandate(Mandate mandate, GoCardlessCustomer customer) {
+    public Mandate createMandate(Mandate mandate, GoCardlessCustomer customer) {
```

Comment:
Maybe this method should be `void` and called something other than `createMandate` so that it’s obvious it’s modifying the passed-in `Mandate` rather than creating something new?

I suppose this method _could_ return an entirely new `Mandate` — I’m not sure what will make it clearer than the `Mandate` needs to be persisted (whether it’s the modified original or a new one).

Part of me is now wondering whether we should have some sort of inheritance relationship whereby there’s a `Mandate` that has a `MandateThatExistsWithAPaymentProvider` (stupid name just to illustrate the point).

Reply:
Yes its a good point and something that I'm not happy with myself. If the only thing that needed to be persisted on our mandate after the creation of the mandate with GoCardless was their Id, then we could make `createMandate` simply return the `PaymentProviderMandateId` back to the MandateService for it to persist (which I think is more intuitive). I'm not sure why we take the bank statement reference at the same time we create the mandate with go cardless as I suspect they take it from something we give them and which we might know earlier in the process. I'll have a look around and see if I can make it cleaner.

Reply:
The reason the bank statement reference comes from GoCardless is because they generate it for us (the Bacs rules for what’s valid are complicated so we let them worry about getting it right).

Reply:
👍 yep, I got that from their docs